packages/backend/src/open_payments/payment/incoming/model.ts

```diff
@@ -165,4 +165,41 @@ export class IncomingPayment
 
     return data
   }
+
+  $formatJson(json: Pojo): Pojo {
+    json = super.$formatJson(json)
+    return {
+      id: json.id,
+      accountId: json.accountId,
+      incomingAmount: this.incomingAmount
+        ? {
+            ...json.incomingAmount,
+            value: json.incomingAmount.value.toString()
+          }
+        : null,
+      receivedAmount: {
+        ...json.receivedAmount,
+        value: json.receivedAmount.value.toString()
+      },
+      state: json.state.toLowerCase(),
+      description: json.description,
+      externalRef: json.externalRef,
+      createdAt: json.createdAt,
+      updatedAt: json.updatedAt,
+      expiresAt: json.expiresAt.toISOString()
+    }
+  }
+}
+
+export type IncomingPaymentJSON = {
+  id: string
+  accountId: string
+  incomingAmount: AmountJSON | null
+  receivedAmount: AmountJSON
+  state: string
+  description: string | null
+  externalRef: string | null
```

Comment:
```suggestion
  incomingAmount?: AmountJSON
  receivedAmount: AmountJSON
  state: string
  description?: string
  externalRef?: string
```
Ditto for outgoing payments and quotes

Reply:
I wouldn't want to call the type `IncomingPaymentJSON` in that case since `undefined` is not valid JSON. What do you think?

Reply:
This makes me wonder if we could get away with not defining `IncomingPaymentJSON` (just specify `$formatJson`)
https://vincit.github.io/objection.js/api/model/overview.html#model-data-lifecycle
>Note: Most libraries like [express](http://expressjs.com/en/index.html)[ (opens new window)](http://expressjs.com/en/index.html) and [koa](https://koajs.com/)[ (opens new window)](https://koajs.com/) automatically call the [toJSON](https://vincit.github.io/objection.js/api/model/instance-methods.html#tojson) method when you pass the model instance to methods like response.json(model). You rarely need to call [toJSON()](https://vincit.github.io/objection.js/api/model/instance-methods.html#tojson) or [$toJson()](https://vincit.github.io/objection.js/api/model/instance-methods.html#tojson) explicitly.

Reply:
I guess that's tricky because `$formatJson` would need `config.publicHost`

Reply:
I guess we could find a hacky way to get it in there but I don't know if we want to do that.

Reply:
What if we removed the `null`s and renamed as `IncomingPaymentResponse`?

Reply:
The thing is that `toJSON()` automatically transforms `undefined` into `null`. That is why I think it makes sense to keep the `null`s and the `IncomingPaymentJSON` name and removing the `null`s from the response the way I do it.